### PR TITLE
Various Slaughter Demon Nerfs

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

Bloodcrawl now has a 5 second cooldown AFTER ENTERING the crawl state, in which they must stay and cannot re-enter the living world.
Slaughter demon code reworked to now no longer qdel, instead store the bodies until they die, and spew it out on death, just like morphs.

## Why It's Good For The Game

I believe the ability for slaughter/laughter demons to blitz attack and overwhelm a group of armed people by rapidly appearing, attacking, and disappearing is a little too strong. This cooldown now means that they will need to strategize their attacks better, and make their punches count. __This does not prevent them from re-entering bloodcrawl, so it isn't negatively impacting their suitability, just their max strength at the hands of an experienced player.__

We're also moving away from outright round removal and qdeleting bodies should probably go away.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Dropping bodies on death
image
image

Cooldown in action
image

</details>

## Changelog
:cl:
balance: Bloodcrawl now forces you to be in it for at least 5 seconds after entering it.
balance: Now, all slaughter demons drop bodies on death instead of deleting them.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
So what's the point of a slaughter demon now compared to a laughter demon?

**Comment by MarkusLarsson421** [comment]
Their blood crawl nerf is too much, I like how it is. Even if it is as OP as it is, they still die very easily. And someone experienced will cleverly use space cleaner smoke to remove all sources of blood. Seeing as how absolutely rare it is, I think it is a good thing to let them have that freedom.

**Comment by Fronsis** [comment]
Thank you! Slaughter demon is one of the few nasty qdels that are still around (another one is #11416 )

> So what's the point of a slaughter demon now compared to a laughter demon?

That has been the main point of discussion everytime we've brought up Slaughter Demon ''nerf'' regarding their qdel, Bee has been in track with removing mechanics that do qdel like Morph, Dragon among others as OP said so..

I'd say the best ''outcome'' that i think is more fitting is:

- Slaughter demon on death gives back the bodies husked(like morph) or without blood or something that's relevant to slaughter(bodies come back with random missing limbs up to 2 max, etcetera)

- Laughter demon on death gives back the bodies without anything happening to them, thus keeping their ''generous'' clown/prank nature

But if we're honest laughter demons are even MORE rare than the slaughter demon and there's no reason to pick one over the another iirc, at this point they could just simply go away.


> Their blood crawl nerf is too much, I like how it is. Even if it is as OP as it is, they still die very easily. And someone experienced will cleverly use space cleaner smoke to remove all sources of blood. Seeing as how absolutely rare it is, I think it is a good thing to let them have that freedom.

I'll have to half agree and half disagree, you're right that 20 seconds is too much, perhaps it should be ten to be a little more generous because you're quite right that once the crew is aware it just take some space cleaners and mop to deny heavily populated areas(like medbay) and they'll need to target isolated areas that happened to have a drop of blood, which i think it's fine, on the other hand it's extremely... unfun to play against one because they have a very easy 1-click escape tool, so they can very easily disengage whenever they feel like it and take their victims to RR them, not to mention they're not **That** fragile it takes quite a bit of laser shots to kill them and most interactions i've had they've easily escaped due to their powers.

**Comment by MarkusLarsson421** [comment]
> they're not That fragile it takes quite a bit of laser shots to kill them and most interactions i've had they've easily escaped due to their powers.

But the ability to heal is very difficult, as you need to find bodies to feed on. More difficult if there isn't any blood in the morgue or other random bodies about.

**Comment by mystery3525** [comment]
> So what's the point of a slaughter demon now compared to a laughter demon?

Laughter demon Aheals on death

**Comment by mystery3525** [comment]
> I'll have to half agree and half disagree, you're right that 20 seconds is too much, perhaps it should be ten to be a little more generous because you're quite right that once the crew is aware it just take some space cleaners and mop to deny heavily populated areas(like medbay) and they'll need to target isolated areas that happened to have a drop of blood, which i think it's fine, on the other hand it's extremely... unfun to play against one because they have a very easy 1-click escape tool, so they can very easily disengage whenever they feel like it and take their victims to RR them, not to mention they're not That fragile it takes quite a bit of laser shots to kill them and most interactions i've had they've easily escaped due to their powers.

Tossed around about this for a bit and thought that 20 seconds is a little too much, but 15 seconds is the lowest I will go. I originally started with 30 seconds, and found that even that is *too* punishing.

**Comment by PowerfulBacon** [comment]
> But if we're honest laughter demons are even MORE rare than the slaughter demon and there's no reason to pick one over the another iirc

Slaughter demons 2 wizard points while laughter demons are 1. Laughter demons are usually always picked by wizards so the only time slaughter demons really appear is when the super rare random event fires to spawn one.

File: code/modules/spells/spell_types/bloodcrawl.dm [hunk 1]
```diff
@@ -23,14 +24,20 @@
...
+	if(phased)
+		if(!COOLDOWN_FINISHED(src, resurface_cooldown))
+			to_chat(user, "<span class='warning'>You need to wait until you can resurface!</span>")
+			return
+		if(user.phasein(target))
+			phased = FALSE
+	else
+		if(user.phaseout(target))
+			phased = TRUE
+			COOLDOWN_START(src, resurface_cooldown, 15 SECONDS)
```
Comment (Rukofamicom) [inline code/modules/spells/spell_types/bloodcrawl.dm]:
```suggestion
			COOLDOWN_START(src, resurface_cooldown, 5 SECONDS)
```

Start small, we can increase it more later if it's still necessary, but even a 5 second cooldown will be a major nerf to the current gamer strats with the demons. We don't want crew to have enough time to completely clean the room or else it becomes way too easy to completely shut the demon down.

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
I think the cooldown is way too long right now, any nerf to their ability to pop in and out will help prevent it from being as much of an issue. I was also observing the round that I believe triggered this PR and though the demons were way too strong when utilizing that. 

(Non-blocking, you just happen to be on the code anyway) Coincidentally, would you mind adding a check that prevents demons from pulling megafauna into the depths? We nearly had a player dump megafauna on the station last night before I yoinked them back into the station and swapped who was controlling it.

**Comment by Rukofamicom** [comment]
I would almost like to see a cooldown on both exiting and re-entering the blood, but it would need to be coupled with the ability for the demon to generate its own blood to pop into and out of. 

Adding a cooldown to the demon on either side greatly increases the risk of crew successfully cleaning all blood which is just a very boring shutdown to the demon.

**Comment by mystery3525** [comment]
> I think the cooldown is way too long right now, any nerf to their ability to pop in and out will help prevent it from being as much of an issue. I was also observing the round that I believe triggered this PR and though the demons were way too strong when utilizing that.

> I would almost like to see a cooldown on both exiting and re-entering the blood, but it would need to be coupled with the ability for the demon to generate its own blood to pop into and out of.

I don't believe there should be a cool down on retreating into the bloodcrawl as that will allow the demon to be killed more easily. Being killed is permanent whereas being cleaned out and not being able to get more kills is temporary, someone is bound to slip up somewhere on the station and bleed.

I still think that the cooldown should be longer considering how easy it is to bleed now with the wounds changes. If the crew is able to successfully clean an area tough luck, it won't stay clean forever. They are just going to have to find other victims outside of medbay (Seems to be pretty much where they hang out most of the round).

**Comment by Rukofamicom** [comment]
If 5 seconds still isn't enough it can be turned up more later.

5 seconds is already a significant nerf to popping in and out repeatedly, the strategy you're trying to prevent

**Review by Rukofamicom — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
Update the CL

**Comment by mystery3525** [comment]
It was already up to date

**Comment by Rukofamicom** [comment]
(I updated it)

**Pull request merged**